### ISSUE TITLE
Type checker fails to report when combining Omit with & {[key: string]: unknown}

### ISSUE THREAD

**Issue by kdeberk** [microsoft/TypeScript#54214]
# Bug Report

### 🔎 Search Terms

omit type checker

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about Omit

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/LAKALgngDgpgBAZTAQzAVwM5wLxwEQDGATjKjACZ5wA++ByAdgTADYV6iiSyJhFoEwAQRxwA3gF84AMnEBtANYwIALjgY+ASwYBzALpq0DBQwD2AdwYTOIbWBhEAZsma9+ggEJwYAD3sNyLCR3YXFQOAj1FHQMNSRUTHDIgDcHDE1TBjUGNABbACMHJIj85CI1DSJtHVBrEFA7B2dXYIEwAGFvPxgArAB5XM0wAB5WzwAafFKiPAA+cTrQAkyNOFU3Ns7cMWKohNi6RmYWNkpJuAB6C7hhgFo4QY1oCjh6JlYKXdSidMy1PEq1SokUu1zucHMREyOjg3BgtSAA)

### 💻 Code

```ts

type Status = "created" | "canceled"

type StructA = {} & {[key: string]: unknown}

interface StructB extends StructA {
    status: Status
    version: number
    bar: string
}

interface StructC extends Omit<StructB, "bar"> {}

const x: StructC = {
    status: "cancelled",  // <- mistyped canceled
    version: "string"     // <- wrong type
}
```

### 🙁 Actual behavior

TS playground is not complaining about the two issues at the bottom. `cancelled` should be `canceled` and `version` should only accept numbers, not strings.

### 🙂 Expected behavior

TS playground should report about the two issues. In case of the cancelled<->canceled issue we're really relying on TS to catch these easy mistakes.

**Comment by jcalz** [comment]
Duplicate of #50638

**Comment by MartinJohns** [comment]
Also, be aware that indexer types with properties are never really type safe. Your type says **any** property has the type `unknown`, so you can assign any type to any property, including `status` and `version`.
